HeterogeneousCore/CUDATest/plugins/TestPortableAnalyzer.cc

```diff
@@ -66,13 +66,17 @@ class TestPortableAnalyzer : public edm::stream::EDAnalyzer<> {
     edm::LogInfo msg("TestPortableAnalyzer");
     msg << source_.encode() << ".size() = " << view.metadata().size() << '\n';
     msg << "  data  @ " << product.buffer().get() << ",\n"
-        << "  x     @ " << view.metadata().addressOf_x() << " = " << Column(view.x(), view.metadata().size()) << ",\n"
-        << "  y     @ " << view.metadata().addressOf_y() << " = " << Column(view.y(), view.metadata().size()) << ",\n"
-        << "  z     @ " << view.metadata().addressOf_z() << " = " << Column(view.z(), view.metadata().size()) << ",\n"
-        << "  id    @ " << view.metadata().addressOf_id() << " = " << Column(view.id(), view.metadata().size()) << ",\n"
-        << "  r     @ " << view.metadata().addressOf_r() << " = " << view.r() << '\n'
-        << "  flags @ " << view.metadata().addressOf_flags() << " = " << Column(view.flags(), view.metadata().size())
+        << "  x     @ " << view.metadata().addressOf_x() << " = " << Column(view.x().data(), view.metadata().size())
```

Comment:
can you add a constructor for Column that takes a span as a single parameter, and change the code here to use it ?